portal/app/api/v1/endpoint/farm_device.py

```diff
@@ -1,8 +1,10 @@
-from rest_framework import (viewsets, serializers)
+from rest_framework import (viewsets, serializers, filters, status)
 from rest_framework.authentication import TokenAuthentication
 from app.api.v1 import CsrfExemptAuth
 from app.common.enum_field_handler import EnumFieldSerializerMixin
-from app.models import FarmDevice
+from app.models import FarmDevice, FarmData
+from rest_framework.decorators import action
+from rest_framework.response import Response
```

Comment:
This action and response aren't being used are they? I think we can get rid of them